### PR TITLE
Disable Context Propagation and useless IDLE connections Netty detection

### PR DESCRIPTION
**Pull request by franz1981**
@cescoffier as well 

Any other feedback?

### CODE REVIEW

**Comment by LesnyRumcajs** [comment]
@franz1981 Could you please briefly elaborate on those flags? I would like to know if they do not take advantage of the logic simplicity and are generic enough - in other words, would you use the same flags for a regular backend application? If so, is there a reason they are not the default?

**Review by cescoffier — APPROVED** [review]
I would also test with the new gRPC server (using the http server)

**Comment by franz1981** [comment]
> you use the same flags for a regular backend application? If so, is there a reason they are not the default?

3 different sets of changes here:
- disable context propagation ( https://quarkus.io/guides/context-propagation) : this is an advanced feature and right now we don't disable it by default although most users shouldn't require it, because we are not validating it at compile time (and I have no idea if is possible). This is the comment from one of the implementor -> https://github.com/quarkusio/quarkus/pull/34132#issuecomment-1597254947
- disable mutiny callback decorators: such are still involved into the context propagation dance. I'm disabling them just in case the former setting isn't enough
- remove the Idle connection handle of Netty: if peer clients/server aren't shield by proxies and TCP FIN work as expected, there's no point for this

So, in short: assuming what the other servers are doing (they do have something to detect half-dead connections/inactive from long time?) i don't see any reason why pay for them.

Said that, I didn't yet profiled the test nor inspected the load generator itself yet :)

**Comment by franz1981** [comment]
And I've noticed another thing: https://quarkus.io/guides/grpc-service-implementation#scaling
The default number of quarkus grpc servers seems to be just 1 (vertx verticle) - meaning that what you have observed so far was probably using a single core?

**Comment by LesnyRumcajs** [comment]
> And I've noticed another thing: https://quarkus.io/guides/grpc-service-implementation#scaling The default number of quarkus grpc servers seems to be just 1 (vertx verticle) - meaning that what you have observed so far was probably using a single core?

I do believe it was addressed in https://github.com/LesnyRumcajs/grpc_bench/pull/361, right?

**Review by LesnyRumcajs — APPROVED** [review]
Thanks for the write-up; I really appreciate it!

**Comment by franz1981** [comment]
> I do believe it was addressed in #361, right?

Nope. Quarkus engine is using Vertx under the hood (which uses Netty), but th pr you have mentioned is on Vertx alone, not Quarkus, which have its own configuration parameters and slightly different default behaviours.

**Comment by LesnyRumcajs** [comment]
Sounds good, thanks!

**Comment by He-Pin** [comment]
@LesnyRumcajs Would you rerun the bench on the same box some later time :) ?

**Comment by LesnyRumcajs** [comment]
> @LesnyRumcajs Would you rerun the bench on the same box some later time :) ?

I'll re-run it this or the next weekend.

**Comment by franz1981** [comment]
any news @LesnyRumcajs ?
